packages/react-graphql/src/hooks/query.ts

```diff
@@ -38,12 +38,16 @@ export default function useQuery<
   } = options;
   const client = useApolloClient(overrideClient);
 
-  if (typeof window === 'undefined' && skip) {
+  if ((typeof window === 'undefined' && skip) || fetchPolicy === 'no-cache') {
```

Comment:
this early return actually breaks [react hook rule](https://github.com/Shopify/quilt/issues/763). Is there anyway not to do this?

Reply:
The rules of hooks don't matter for server rendering. They only exist so that a single component, when updated, calls hooks in the same order, but in server renders, the entire component is thrown out each time.

Reply:
that make sense to me, so this is really a dev experience thing to for consumer of this package to understand why they would be getting this warning.